aztokenprovider/token_provider.go

```diff
@@ -140,8 +140,14 @@ func (c *managedIdentityTokenRetriever) GetCacheKey() string {
 }
 
 func (c *managedIdentityTokenRetriever) Init() error {
-	options := &azidentity.ManagedIdentityCredentialOptions{ID: azidentity.ClientID(c.clientId)}
-	if credential, err := azidentity.NewManagedIdentityCredential(options); err != nil {
+	var credential *azidentity.ManagedIdentityCredential
+	var err error
```

Comment:
you don't need to define these two variables in advance now :)

Reply:
Fixed 😊 On your above comment - I'm a bit confused as the `credential` interface is already a part of the `managedIdentityTokenRetriever` struct. We can discuss this offline and I'll add a test at a later date 😊